worlds/jakanddaxter/Rules.py

```diff
@@ -22,48 +21,47 @@ def set_orb_trade_rule(world: JakAndDaxterWorld):
 
     if options.enable_orbsanity == EnableOrbsanity.option_off:
         world.can_trade = lambda state, required_orbs, required_previous_trade: (
-            can_trade_vanilla(state, player, required_orbs, required_previous_trade))
+            can_trade_vanilla(state, player, world, required_orbs, required_previous_trade))
     else:
         world.can_trade = lambda state, required_orbs, required_previous_trade: (
-            can_trade_orbsanity(state, player, required_orbs, required_previous_trade))
+            can_trade_orbsanity(state, player, world, required_orbs, required_previous_trade))
 
 
-def recalculate_reachable_orbs(state: CollectionState, player: int) -> None:
+def recalculate_reachable_orbs(state: CollectionState, player: int, world: JakAndDaxterWorld) -> None:
 
     if not state.prog_items[player]["Reachable Orbs Fresh"]:
 
         # Recalculate every level, every time the cache is stale, because you don't know
         # when a specific bundle of orbs in one level may unlock access to another.
         for level in level_table:
             state.prog_items[player][f"{level} Reachable Orbs".strip()] = (
-                count_reachable_orbs_level(state, player, state.multiworld, level))
+                count_reachable_orbs_level(state, world, level))
 
         # Also recalculate the global count, still used even when Orbsanity is Off.
-        state.prog_items[player]["Reachable Orbs"] = count_reachable_orbs_global(state, player, state.multiworld)
+        state.prog_items[player]["Reachable Orbs"] = count_reachable_orbs_global(state, world)
         state.prog_items[player]["Reachable Orbs Fresh"] = True
 
 
 def count_reachable_orbs_global(state: CollectionState,
-                                player: int,
-                                multiworld: MultiWorld) -> int:
+                                world: JakAndDaxterWorld) -> int:
 
     accessible_orbs = 0
     # Cast all regions upfront to access their unique attributes.
-    for region in typing.cast(typing.List[JakAndDaxterRegion], multiworld.get_regions(player)):
-        # Rely on short-circuiting to skip region.can_reach whenever possible.
-        if region.orb_count > 0 and region.can_reach(state):
-            accessible_orbs += region.orb_count
+    for level_name in world.level_to_regions:
+        for region in world.level_to_regions[level_name]:
```

Comment:
up to you i just find this a bit easier to read
```suggestion
    for level_regions in world.level_to_regions.values():
        for region in level_regions:
```